lib/ueberauth/strategy/microsoft/oauth.ex

```diff
@@ -42,14 +49,21 @@ defmodule Ueberauth.Strategy.Microsoft.OAuth do
     |> AuthCode.get_token(params, headers)
   end
 
+  def refresh_token(client, params, headers) do
+    client
+    |> put_param(:client_id, client.client_id)
+    |> put_param(:client_secret, client.client_secret)
+    |> Refresh.get_token(params, headers)
+  end
+
   defp defaults(config) do
     tenant_id = config[:tenant_id] || "common"
 
     [
       strategy: __MODULE__,
-      site: "https://graph.microsoft.com",
+      site: "https://login.microsoftonline.com/#{tenant_id}/oauth2/v2.0",
```

Comment:
I think `https://graph.microsoft.com` is the correct value for `site` as this describes the site we're authenticating to get access to. The `https://login.microsoftonline.com/` URL is only used for carrying out the authentication.

Reply:
Tokens are refreshed by submitted a POST request to  /token of the endpoint, which is  "https://login.microsoftonline.com/#{tenant_id}/oauth2/v2.0". The OUATH refresh strategy concats the site and token opts opts to derive the endpoint URI.

The "https://graph.microsoft.com" endpoint is for the GraphAPI itself after authentication.